prov/rxd/src/rxd_cq.c

```diff
@@ -1064,6 +1065,7 @@ static void rxd_handle_start_data(struct rxd_ep *ep, struct rxd_peer *peer,
 repost:
 	rxd_ep_repost_buff(rx_buf);
 out:
+	assert(rxd_reposted_bufs);
```

Comment:
I'm guessing based on the function names that rxd_ep_repost_buff() will increment rxd_reposted_bufs.  Is that correct?  If so, my question is whether that reposted buffer can be used between rxd_ep_repost_buff returning and the assert check.  Does the code have the right locking to protect against this, or is there the potential for a (small) race here?